catroid/src/main/java/org/catrobat/catroid/sensing/CollisionInformation.java

```diff
@@ -369,10 +377,12 @@ public static Polygon createPolygonFromPoints(ArrayList<PointF> points) {
 	public static boolean[][] createCollisionGrid(Bitmap bitmap) {
 		int width = bitmap.getWidth();
 		int height = bitmap.getHeight();
+		int[] pixels = new int[width * height];
+		bitmap.getPixels(pixels, 0, width, 0, 0, width, height);
 		boolean[][] grid = new boolean[width][height];
 		for (int x = 0; x < width; x++) {
 			for (int y = 0; y < height; y++) {
-				if (bitmap.getPixel(x, y) != 0) {
+				if (pixels[y * width + x] != 0) {
```

Comment:
what is the advantage of this over accessing via bitmap.getPixel? this is way harder to read and understand whats going on.

Reply:
it seems always accessing the bitmap is more troublesome than accessing the array and while testing (and trying to improve the performance) this change helped to speed up the calculation a bit. It was about 2 seconds less computation time, but it's not so much since the computation takes 20-30 seconds anyway.